components/brave_wallet/browser/keyring_service.cc

```diff
@@ -626,7 +645,9 @@ HDKeyring* KeyringService::RestoreKeyring(const std::string& keyring_id,
     return nullptr;
   }
 
-  if (!CreateKeyringInternal(keyring_id, mnemonic, is_legacy_brave_wallet)) {
+  // non default keyrings can only create encryptors for lazily keyring creation
+  if (keyring_id != mojom::kDefaultKeyringId ||
```

Comment:
mind to add a test to check the keyring observers will be called only for the default keyring?

Reply:
good call, fixed in cd6bbd54c7b468a741e9a7af2954bd31ac68ec58